sh/tests/joker-num_test.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
```

Comment:
To make sure the tester exit at the first failed test, add the following:
```bash
set -e 
```

the `-o pipefail` should not add any value in this case as the `-u` for undefined variables

Reply:
I've added the full unofficial strict mode syntax